### PR TITLE
added update dto

### PR DESCRIPTION
**Pull request by Ziron99**
added update dto for updating product data

### CODE REVIEW

File: src/main/java/com/festy/app/Festy/product/ProductController.java [hunk 1]
```diff
@@ -42,6 +47,14 @@ public String saveProduct(@ModelAttribute("product") Product product, @RequestPa
         service.addProductWithImage(product,file);
         return "redirect:/product";
     }
+
+    @PostMapping("/update")
+    public String saveProduct(@ModelAttribute("product") ProductUpdateDTO updateDTO) {
+        service.uppdateProduct(updateDTO);
+//        System.out.println(updateDTO);
```
Comment (zander-b) [inline src/main/java/com/festy/app/Festy/product/ProductController.java]: - [x] please remove

File: src/main/java/com/festy/app/Festy/product/ProductController.java [hunk 2]
```diff
@@ -26,14 +24,21 @@ public String getDashboard(Model model){
         return "ProductDashboard";
     }
 
+    private ProductUpdateDTO convertResponse(Product product) {
```
Comment (zander-b) [inline src/main/java/com/festy/app/Festy/product/ProductController.java]: Is this used anywhere?

File: src/main/java/com/festy/app/Festy/product/ProductServices.java [hunk 3]
```diff
@@ -46,4 +46,20 @@ public Product getProductById(long id){
     public void deleteProduct(Long id){
         repository.deleteById(id);
     }
+
+    public void uppdateProduct(ProductUpdateDTO productUpdateDTO) {
+        Optional< Product > optional = repository.findById(productUpdateDTO.getId());
+        Product product = null;
+        if (optional.isPresent()) {
```
Comment (zander-b) [inline src/main/java/com/festy/app/Festy/product/ProductServices.java]: Please remove the if statement
and change 
`Product product = null `
to 
`Product product = optional.orElseThrow(() -> new RuntimeException("product not found for id"+productUpdateDTO.getId()))`

Comment (Ziron99) [reply src/main/java/com/festy/app/Festy/product/ProductServices.java]: done

**Review by zander-b — APPROVED** [review]

**Pull request merged**